apps/site/lib/site_web/channels/vehicle_channel.ex

```diff
@@ -51,14 +62,14 @@ defmodule SiteWeb.VehicleChannel do
           id: vehicle.id,
           icon: "vehicle-bordered-expanded",
           rotation_angle: vehicle.bearing,
-          shape_id: trip.shape_id,
+          shape_id: preprocessed_trip.shape_id,
```

Comment:
Following up on my comment above, using an inverse of the same trick you can also do something like `trip && trip.shape_id` here and avoid having to make the `preprocessed_trip` in the first place.